corehq/apps/userreports/tasks.py

```diff
@@ -238,9 +239,27 @@ def delete_data_source_task(domain, config_id):
     delete_data_source_shared(domain, config_id)
 
 
-@periodic_task(serializer='pickle',
-    run_every=crontab(minute='*/5'), queue=settings.CELERY_PERIODIC_QUEUE
-)
+@periodic_task(serializer='pickle', run_every=crontab(minute='*/5'), queue=settings.CELERY_PERIODIC_QUEUE)
+def reprocess_archive_stubs():
+    # Exit this task after 1 minute so that the same stub isn't ever processed in multiple queues.
+    check_repeaters()
```

Comment:
I think there was a misunderstanding here. This was an example of a task that exits after a certain time period. Calling it here won't cause this task to exit and it will also fire off all the repeaters scheduled to run. You need to do something like this:
```python
start = time.time()
for .....:
    if time.time() - start > cutoff:
        return
```

Reply:
My mistake: https://github.com/dimagi/commcare-hq/pull/22470/commits/0ae907560cff5e22e900a6146f2c5c930433cab2